### PR TITLE
Use jetstack public to build and fix Application wrangling

### PR DESCRIPTION
**Pull request by maelvls**
In this PR, I fix a couple of things

2. Remind ourselves that we should only use `jetstack-public` in order to run `gcloud builds submit`, otherwise the verify step will fail (see "hardcoded" in cloudbuild.yml)
3. Fix the Application component wrangling (cainjector and webhook still missing...). the names are confusion thourh (controller = google-cas-issuer), so we need to work that out
4. ubbagent is now a side-car to the cert-manager controller deployment
5. I had to unpack the cert-manager tarball... sorry for the 16 000 new lines!!!

### CODE REVIEW

**Comment by maelvls** [comment]
Since I had to push an image in order to get our application reviewed ASAP, I pushed `1.1.0-gcm.2`. The steps were:

```sh
% gcloud builds submit --project jetstack-public --timeout 1800s --config cloudbuild.yaml \
  --substitutions _CLUSTER_NAME=smoke-test,_CLUSTER_LOCATION=europe-west2-b,_APP_MINOR_VERSION=1.1,_APP_VERSION=1.1.0-gcm.2
Logs are available at [https://console.cloud.google.com/cloud-build/builds/7a96460d-dcc5-47bd-be97-010f99fe56bf?project=885059085598].
```

Then I retagged all images:

```sh
retag() {
  docker pull $1 && docker tag $1 $2 && docker push $2
}
retag gcr.io/jetstack-public/jetstack-secure-for-cert-manager:{google-review,1.1.0-gcm.2}
retag gcr.io/jetstack-public/jetstack-secure-for-cert-manager/cert-manager-acmesolver:{google-review,1.1.0-gcm.2}
retag gcr.io/jetstack-public/jetstack-secure-for-cert-manager/cert-manager-cainjector:{google-review,1.1.0-gcm.2}
retag gcr.io/jetstack-public/jetstack-secure-for-cert-manager/cert-manager-webhook:{google-review,1.1.0-gcm.2}
retag gcr.io/jetstack-public/jetstack-secure-for-cert-manager/cert-manager-google-cas-issuer:{google-review,1.1.0-gcm.2}
retag gcr.io/jetstack-public/jetstack-secure-for-cert-manager/preflight:{google-review,1.1.0-gcm.2}
retag gcr.io/cloud-marketplace-tools/metering/ubbagent:latest gcr.io/jetstack-public/jetstack-secure-for-cert-manager/ubbagent:1.1.0-gcm.2
```

Then I updated the version on the UI and clicked "Submit for review".

**Comment by maelvls** [comment]
I have cut another release, `1.1.0-gcm.3`. With this last release, we fixed all the issues that we knew of (icon is now showing, deployment names are now way more descriptive, ubbagent is integrated in the main cert-manager pod instead of being a standalone deployment).

My plan is to click "Submit for review" in a few minutes, and have Google review `1.1.0-gcm.3` during the night (11-12 March). If Google accepts it during the night, we should be able to make the application public by tomorrow morning! 

<details>
<summary>step by step process</summary>

first I built `1.1.0-gcm.3` deployer

```
% gcloud builds submit --project jetstack-public --timeout 1800s --config cloudbuild.yaml \
  --substitutions _CLUSTER_NAME=smoke-test,_CLUSTER_LOCATION=europe-west2-b,_APP_MINOR_VERSION=1.1,_APP_VERSION=1.1.0-gcm.3
```

- [logs](https://console.cloud.google.com/cloud-build/builds/4dde7620-4a06-434c-9dc9-71ed8c6d7dff?project=885059085598)
- image: gcr.io/jetstack-public/jetstack-secure-for-cert-manager/deployer:1.1.0-gcm.3 (+2 more)  SUCCESS

finally I did

```sh
git tag 1.1.0-gcm.3
```

</details>

File: chart/jetstack-secure-gcm/charts/preflight/templates/_helpers.tpl
Comment (wallrj) [inline chart/jetstack-secure-gcm/charts/preflight/templates/_helpers.tpl:51]: nit. Are we calling it preflight, or are we calling it `jetstack-secure-agent`, or somthing else? 
I guess preflight makes sense until we change the name of the chart.

File: chart/jetstack-secure-gcm/templates/billing-agent-config.yml
Comment (wallrj) [inline chart/jetstack-secure-gcm/templates/billing-agent-config.yml:11]: Can I wondered if we could use more descriptive values for these component names, e.g. `Google Cloud Marketplace Billing Agent`....

BUT WAIT why is this still here as a separate deployment, if you have now deployed the ubbagent as a cert-manager side-car?

Comment (wallrj) [reply chart/jetstack-secure-gcm/templates/billing-agent-config.yml:11]: ...ah, this isn't the deployment, but the config for ubbagent...ignore me. 
I see the deployment is deleted below.

**Review by wallrj — APPROVED** [review]
Thanks @maelvls 

Just a few comments and nits, but I know that this has already been deployed, so I'll add lgtm and hold so that you can decide when it gets merged.

/lgtm
/hold

File: README.md
Comment (maelvls) [reply README.md:193]: I remember about having to separate behaviours:
- When running the smoke-test, we set replica=0
- When running normally (e.g., one click deployment through the marketplace UI), the preflight deployment would be set to replica=1 (= purposefully failing)

I don't really like the idea of letting the preflight deployment in a failing mode "by default", e.g. the user would immediately see a failing deployment after deploying:

The test-1 application page on GKE should show the test-1 application. The preflight deployment is failing because the user has not (yet) gone to http://platform.jetstack.io/ to register their cluster. This screenshot is stored in this issue: https://github.com/jetstack/jetstack-secure-gcm/issues/21

File: chart/jetstack-secure-gcm/templates/application.yaml
Comment (maelvls) [reply chart/jetstack-secure-gcm/templates/application.yaml:13]: had to dig into [click-to-deploy](https://github.com/GoogleCloudPlatform/click-to-deploy/blob/f02b0b927bf635d9e65871b71866d456a3da41e7/k8s/cert-manager/chart/cert-manager/templates/application.yaml#L7) to find this piece of the Application API 😅

File: chart/jetstack-secure-gcm/charts/preflight/templates/_helpers.tpl
Comment (maelvls) [reply chart/jetstack-secure-gcm/charts/preflight/templates/_helpers.tpl:51]: I am quite confused by this naming unknown: in places, I call it the "Jetstack Secure Platform agent", and sometimes I default to "the preflight agent".

@charlieegan3 thoughts?

File: README.md
Comment (maelvls) [reply README.md:238]: Initially I tried the SelfSignedIssuer but I could not figure it out... I guess I should double down on that 😅

If that's OK, I will /unhold and do this change in another PR
I'll add that to my todo

File: chart/jetstack-secure-gcm/templates/billing-agent-config.yml
Comment (maelvls) [reply chart/jetstack-secure-gcm/templates/billing-agent-config.yml:11]: Apologies for the messy PR :-(

**Comment by maelvls** [comment]
/unhold

I will create a new PR with the changes after your comments.

File: README.md
Comment (maelvls) [reply README.md:193]: I'll open an issue about this.

**Pull request merged**
